### PR TITLE
update: Lottery Percentage to Gov Accounts

### PR DESCRIPTION
**Pull request by Savage912-Official**
Once the winner has been chosen half of the pot will auto be put into a random government account.

Also redid the config to look a bit cleaner than it did.

### CODE REVIEW

File: resources/usa_lottery/server/sv_lottery.lua [hunk 1]
```diff
@@ -21,10 +21,19 @@ RegisterNetEvent("usa_lottery:choosewinner", function()
     local winningticket = math.random(1, 500)
     local check = 'placeholder'
     local result = MySQL.query.await('SELECT * FROM lotterytotal where lotto = ?', {check})
-    local amount = result[1].total
+    local amount = result[1].total/2
```
Comment (minipunch) [inline resources/usa_lottery/server/sv_lottery.lua:24]:
```suggestion
    local GOV_PERCENTAGE = 0.20 -- preferably at the top of the file
    local govAmount = `GOV_PERCENTAGE * result[1].total`
    local restAmount = result[1].total - govAmount
    -- etc...
```

File: resources/usa_lottery/server/sv_lottery.lua [hunk 2]
```diff
@@ -104,3 +113,23 @@ RegisterServerCallback {
...
+            if doc then
+                --print("* Balance updated in DB! *")
+            end
+		end)
+	end)
+end
+
+function GetCurrentBalance(ident, cb, fundAccount)
+    TriggerEvent('es:exposeDBFunctions', function(db)
+        db.getDocumentById("govfunds", ident, function(doc)
```
Comment (minipunch) [inline resources/usa_lottery/server/sv_lottery.lua:129]:
```suggestion
        local doc = exports.essentialmode:getDocumentById("govfunds", ident)
```
Then get rid of the `es:exposeDBFunctions` event part

Cleaner this way

File: resources/usa_lottery/server/sv_lottery.lua [hunk 3]
```diff
@@ -104,3 +113,23 @@ RegisterServerCallback {
         end
     end
 }
+
+function SaveNewBalance(finishedBalance, fundAccount)
+    TriggerEvent('es:exposeDBFunctions', function(db)
+        db.updateDocument("govfunds", fundAccount, { content = finishedBalance }, function(doc, err, rText)
```
Comment (minipunch) [inline resources/usa_lottery/server/sv_lottery.lua:119]:
```suggestion
        exports.essentialmode:updateDocument("govfunds", fundAccount, { content = finishedBalance })
```

And also get rid of the `es:exposeDBfunctions` event part

File: resources/usa_lottery/server/sv_lottery.lua [hunk 4]
```diff
@@ -21,10 +21,19 @@ RegisterNetEvent("usa_lottery:choosewinner", function()
...
     local check = 'placeholder'
     local result = MySQL.query.await('SELECT * FROM lotterytotal where lotto = ?', {check})
-    local amount = result[1].total
+    local amount = result[1].total/2
     local reset = 1
+    local fundAccount = Config.GovAccounts[math.random(1, #Config.GovAccounts)]
     MySQL.Async.execute("UPDATE lotterytotal SET winner = '"..winningticket.."'")
     TriggerClientEvent("chatMessage", -1, "[^2Los Santos Lottery^0] Lottery winner has been chosen! Winning Number: " .. winningticket .. "\n Head over to Life Invader to claim your winnings.")
+
+    GetCurrentBalance(fundAccount, function(bal)
```
Comment (minipunch) [inline resources/usa_lottery/server/sv_lottery.lua:30]: Will have to update how u use the function if u do my suggestions below

**Review by minipunch — CHANGES_REQUESTED** [review]

File: resources/usa_lottery/server/sv_lottery.lua (hunk 4 above)
Comment (Savage912-Official) [reply resources/usa_lottery/server/sv_lottery.lua:32]: Done

File: resources/usa_lottery/server/sv_lottery.lua [hunk 5]
```diff
@@ -21,11 +21,22 @@ RegisterNetEvent("usa_lottery:choosewinner", function()
     local winningticket = math.random(1, 500)
     local check = 'placeholder'
     local result = MySQL.query.await('SELECT * FROM lotterytotal where lotto = ?', {check})
-    local amount = result[1].total
+    local govAmount = Config.Gov_Percentage * result[1].total
+    local resetAmount = result[1].total - govAmount
```
Comment (minipunch) [inline resources/usa_lottery/server/sv_lottery.lua:25]:
```suggestion
    local restAmount = result[1].total - govAmount
```

File: resources/usa_lottery/server/sv_lottery.lua [hunk 6]
```diff
@@ -21,11 +21,22 @@ RegisterNetEvent("usa_lottery:choosewinner", function()
...
-    MySQL.Async.execute("UPDATE lotterytotal SET total = '"..amount.."'")
+
+    GetCurrentBalance(fundAccount, function(bal)
+        amount = math.abs(govAmount)
+        amount = math.floor(govAmount)
+        local finishedBalance = bal + govAmount
+        SaveNewBalance(finishedBalance, fundAccount)    
+    end, fundAccount)
+
+    MySQL.Async.execute("UPDATE lotterytotal SET total = '"..math.floor(resetAmount).."'")
```
Comment (minipunch) [inline resources/usa_lottery/server/sv_lottery.lua:39]:
```suggestion
    MySQL.Async.execute("UPDATE lotterytotal SET total = '"..math.floor(restAmount).."'")
```

File: resources/usa_lottery/server/sv_lottery.lua [hunk 7]
```diff
@@ -120,4 +131,22 @@ RegisterServerCallback {
             return "Lottery ends in "..math.." days."
         end
     end
-}
\ No newline at end of file
+}
+
+function SaveNewBalance(finishedBalance, fundAccount)
+    TriggerEvent('es:exposeDBFunctions', function(db)
```
Comment (minipunch) [inline resources/usa_lottery/server/sv_lottery.lua:137]: Don't need this when you use `exports.essentialmode...`

File: resources/usa_lottery/server/sv_lottery.lua [hunk 8]
```diff
@@ -120,4 +131,22 @@ RegisterServerCallback {
...
+        exports.essentialmode:updateDocument("govfunds", fundAccount, { content = finishedBalance })
+        if doc then
+            --print("* Balance updated in DB! *")
+        end
+    end)
+end
+
+function GetCurrentBalance(ident, cb, fundAccount)
+    TriggerEvent('es:exposeDBFunctions', function(db)
```
Comment (minipunch) [inline resources/usa_lottery/server/sv_lottery.lua:146]: Same here

**Review by minipunch — CHANGES_REQUESTED** [review]

File: resources/usa_lottery/server/sv_lottery.lua [hunk 9]
```diff
@@ -131,4 +142,18 @@ RegisterServerCallback {
 
         return "Current lottery total is  $"..total.."."
     end
-}
\ No newline at end of file
+}
+
+function SaveNewBalance(finishedBalance, fundAccount)
+    exports.essentialmode:updateDocument("govfunds", fundAccount, { content = finishedBalance })
+    if doc then
```
Comment (minipunch) [inline resources/usa_lottery/server/sv_lottery.lua:149]: Think u forgot to remove this

File: resources/usa_lottery/server/sv_lottery.lua [hunk 10]
```diff
@@ -131,4 +142,18 @@ RegisterServerCallback {
 
         return "Current lottery total is  $"..total.."."
     end
-}
\ No newline at end of file
+}
+
+function SaveNewBalance(finishedBalance, fundAccount)
+    exports.essentialmode:updateDocument("govfunds", fundAccount, { content = finishedBalance })
```
[the remark this answers is not in the document]
Comment (Savage912-Official) [reply resources/usa_lottery/server/sv_lottery.lua:148]: Yeah I’ll add it

[the remark this answers is not in the document]
Comment (Savage912-Official) [reply resources/usa_lottery/server/sv_lottery.lua:148]: I’ll get it done in the morning

File: resources/usa_lottery/server/sv_lottery.lua [hunk 11]
```diff
@@ -21,11 +21,27 @@ RegisterNetEvent("usa_lottery:choosewinner", function()
...
+    -- etc...
     local reset = 1
+    local fundAccount = Config.GovAccounts[math.random(1, #Config.GovAccounts)]
     MySQL.Async.execute("UPDATE lotterytotal SET winner = '"..winningticket.."'")
     TriggerClientEvent("chatMessage", -1, "[^2Los Santos Lottery^0] Lottery winner has been chosen! Winning Number: " .. winningticket .. "\n Head over to Life Invader to claim your winnings.")
-    MySQL.Async.execute("UPDATE lotterytotal SET total = '"..amount.."'")
+
+    GetCurrentBalance(fundAccount, function(bal)
+        amount = math.abs(govAmount)
+        amount = math.floor(govAmount)
```
Comment (minipunch) [inline resources/usa_lottery/server/sv_lottery.lua:34]:
```suggestion
```

not using these `amount` variables anywhere ?

File: resources/usa_lottery/server/sv_lottery.lua [hunk 12]
```diff
@@ -21,11 +21,27 @@ RegisterNetEvent("usa_lottery:choosewinner", function()
...
+    local restAmount = result[1].total - govAmount
+    -- etc...
     local reset = 1
+    local fundAccount = Config.GovAccounts[math.random(1, #Config.GovAccounts)]
     MySQL.Async.execute("UPDATE lotterytotal SET winner = '"..winningticket.."'")
     TriggerClientEvent("chatMessage", -1, "[^2Los Santos Lottery^0] Lottery winner has been chosen! Winning Number: " .. winningticket .. "\n Head over to Life Invader to claim your winnings.")
-    MySQL.Async.execute("UPDATE lotterytotal SET total = '"..amount.."'")
+
+    GetCurrentBalance(fundAccount, function(bal)
+        amount = math.abs(govAmount)
```
Comment (minipunch) [inline resources/usa_lottery/server/sv_lottery.lua:33]:
```suggestion
```

File: resources/usa_lottery/server/sv_lottery.lua [hunk 13]
```diff
@@ -132,3 +148,34 @@ RegisterServerCallback {
         return "Current lottery total is  $"..exports.globals:comma_value(total).."."
     end
 }
+
+function SaveNewBalance(finishedBalance, fundAccount)
+    exports.essentialmode:updateDocument("govfunds", fundAccount, { content = finishedBalance })
+end
+
+function GetCurrentBalance(ident, cb, fundAccount)
```
Comment (minipunch) [inline resources/usa_lottery/server/sv_lottery.lua:154]:
```suggestion
function GetCurrentBalance(ident, cb)
```

not used

**Review by minipunch — APPROVED** [review]

**Pull request merged**